contributors/design-proposals/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,81 @@
+# Priority in ResourceQuota
+
+@resouer
+
+Aug 2017
+  * [Objective](#objective)
+    * [Non-Goals](#non-goals)
+  * [Background](#background)
+  * [Overview](#overview)
+  * [Detailed Design](#detailed-design)
+    * [Expected behavior of ResourceQuota apply to pod](#resourcequota-apply-to-pod)
+    * [Expected behavior of ResourceQuota admission controller](#resourcequota-admission-controller)
+
+
+## Objective
+
+This feature is designed to make ResourceQuota become priority aware, three sub-tasks are included.
+
+* Add a field to ResourceQuota that specifies a priority class name.
```

Comment:
We don't allow changing the integer value of a priority class, but even without it and after thinking more, I guess applying quota by priority class name is more intuitive for users.
I like your suggestion that "PriorityClassName in ResourceQuota is matched against PriorityClassName in the Pod, including the case where PriorityClassName is not set.", but this needs a couple of changes to the original proposal. The rule that "If priority class name is not present, this quota applies to all priority classes." will no longer be valid. We should also allow creation of a resource quota with priority class when a resource quota with no priority class exists.